src/packagedcode/go.py

```diff
@@ -0,0 +1,256 @@
+    if exclude:
+        for name, version in exclude:
+            package_dependencies.append(
+                models.DependentPackage(
+                    purl=PackageURL(
+                        type='gopkg',
+                        name=name,
+                        version=version
+                    ).to_string(),
+                    requirement=version,
+                    scope='exclude',
+                    is_runtime=True,
+                    is_optional=False,
+                    is_resolved=True,
+                )
+            )
+
+    return GoPackage(
+        name=name,
+        homepage_url=homepage_url,
+        download_url=download_url,
+        code_view_url=homepage_url,
+        dependencies=package_dependencies
+    )
+
+
+####################################################################
+# Regrex expressions to parse different types of dependency
+module_parse = re.compile(
+    r'^module\s'
+    r'(?P<module>[a-z].*)'
+).match
+
+module_name_parse = re.compile(
+    r'^module\s.*/'
+    r'(?P<name>[^\s]*)'
+).match
+
+require_parse = re.compile(
+    r'^require\s'
+    r'.*/(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>[a-z].*)'
+).match
+
+exclude_parse = re.compile(
+    r'^exclude\s'
+    r'.*/(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>[a-z].*)'
+).match
+
+dep_link_parse = re.compile(
+    r'^.*/(?P<name>[^\s]*)'
+    r'\s'
+    r'(?P<version>[a-z].*)'
+).match
+
+"""
+https://golang.org/ref/mod#go.mod-files
+
+module is in the form
+module github.com/alecthomas/participle
+>>> p = module_parse('module github.com/alecthomas/participle')
+>>> assert p.group('module') == ('github.com/alecthomas/participle')
+
+require or exclude can be in the form
+require github.com/davecgh/go-spew v1.1.1
+or
+exclude github.com/davecgh/go-spew v1.1.1
+or
+github.com/davecgh/go-spew v1.1.1
+
+>>> p = require_parse('require github.com/davecgh/go-spew v1.1.1')
+>>> assert p.group('name') == ('go-spew')
+>>> assert p.group('version') == ('v1.1.1')
+
+>>> p = exclude_parse('exclude github.com/davecgh/go-spew v1.1.1')
+>>> assert p.group('name') == ('go-spew')
+>>> assert p.group('version') == ('v1.1.1')
+
+>>> p = dep_link_parse('github.com/davecgh/go-spew v1.1.1')
+>>> assert p.group('name') == ('go-spew')
+>>> assert p.group('version') == ('v1.1.1')
+"""
+
+def preprocess(line):
+    """
+    Removes the comment portion and excess spaces.
+    """
+    if "//" in line:
+        line = line[:line.index('//')]
+    line = line.strip()
+    return line
+
+
+def parse_gomod(location):
+    """
+    Method to handle go.mod files.
+    """
+    contents = io.open(location, encoding='utf-8').readlines()
+    gomod_data = {}
+    require = []
+    exclude = []
+
+    for i, line in enumerate(contents):
+        line = preprocess(line)
+        parsed = module_parse(line)
+        if parsed:
+            gomod_data['module'] = parsed.group('module')
+
+        parsed = module_name_parse(line)
```

Comment:
Can these function names start with `parse`? for example: `module_name_parse() => parse_module_name`. This is easier to read/follow

Reply:
I will change this soon:)